### ISSUE TITLE
Implement Solr to improve load times on search views?

### ISSUE THREAD

**Issue by jacobdgm** [DDMAL/CantusDB#435]
At the LinkedMusic project meeting, someone raised the point that fast load times are the feature that everyone wants but no one asks for, and that developers often don't bother trying to fix long load times when performing database searches because they don't know there are ways to make things faster. Solr was suggested as a tool for optimizing search times. Solr is already used on [CantusUltimus](https://github.com/DDMAL/cantus), I believe, so this might be a natural thing to implement.

This will require some amount of research, and its implementation is not urgent - just putting this here so as not to forget about it.

**Comment by fujinaga** [comment]
Yes, this will be good.

**Comment by jacobdgm** [comment]
If I understand correctly, we're trying to use only django builtins for our searching, in order to not add an extra layer of complexity by adding Solr. Closing.

**Comment by fujinaga** [comment]
No, we would like to implement Solr at some point unless all searches are blazingly fast.

**Comment by ahankinson** [comment]
Quite a lot of the search problems you’ve been having have very good solutions in Solr. I’ve been avoiding commenting because I think it would be a distraction at this point to introduce Solr, and not helpful for just getting the site stable and launched. But I would be happy to provide some guidance about this when the time comes.

**Comment by kyrieb-ekat** [comment]
Closing in favor of Django search improvements. May revisit in future.
